lib/fluent/time.rb

```diff
@@ -0,0 +1,73 @@
+module Fluent
+  class NanoTime
+    TYPE = 0
+
+    def initialize(sec, nsec = 0)
+      @sec = sec
+      @nsec = nsec
+    end
+
+    def ==(other)
+      if other.is_a?(Fluent::NanoTime)
+        @sec == other.sec
+      else
+        @sec == other
+      end
+    end
+
+    def sec
+      @sec
+    end
+
+    def nsec
+      @nsec
+    end
+
+    def to_int
+      @sec
+    end
+
+    # for Time.at
+    def to_r
+      Rational(@sec * 1_000_000_000 + @nsec, 1_000_000_000)
+    end
+
+    # for > and others
+    def coerce(other)
+      [other, @sec]
+    end
+
+    def to_s
+      @sec.to_s
+    end
+
+    def to_msgpack(io = nil)
+      @sec.to_msgpack(io)
+    end
+
+    def to_msgpack_ext
+      [@sec, @nsec].pack('NN')
+    end
+
+    def self.from_msgpack_ext(data)
+      new(*data.unpack('NN'))
+    end
+
+    def self.from_time(time)
+      Fluent::NanoTime.new(time.to_i, time.nsec)
+    end
```

Comment:
> `to_time`

I would recommend to use `Time.at(time)` because we can use same code for NanoTime and Integer.

> `self.now`, `self.parse(*args)`

Ah, at first I implemented this methods, but I removed them later to keep simple API.
Now, I agree because I feel `Fluent::NanoTime.from_time(...)` is too verbose.